### ISSUE TITLE
Add `as_dataframe` method for group level design matrix

### ISSUE THREAD

**Issue by drbenvincent** [bambinos/formulae#98]
Simulated data:
```python
groups = 3
samples_per_group = 10
N = groups * samples_per_group
x = np.tile(np.linspace(0, 10, samples_per_group), groups)
group = np.repeat(np.arange(groups), samples_per_group)
y = 2 + -0.5 * group + rng.normal(0, 0.1, N)
df = pd.DataFrame({"x": x, "group": group, "y": y})
```

If we model this with `y ~ 1 + (1 | C(group))` and were interested in the design matrices, then at the moment we _can_ do this:

```python
dm = design_matrices("y ~ 1 + (1 | C(group))", df)
dm.common.as_dataframe()
```

But we cannot call the `as_dataframe` method on the group level design matrix. Ie. we can only call

```python
np.array(dm.group)
```
which result in this
![Screenshot 2023-06-18 at 09 59 31](https://github.com/bambinos/formulae/assets/6765047/cb5863d0-99e4-4f0a-b742-58f37c4b34b1)

but it would be nice to be able to call

```python
dm.group.as_dataframe()
```

Not knowing how the internals work, I believe it would just be a matter of building a dataframe with the correct column labels, which should be `1|C(group)[0]`, `1|C(group)[1]`, and `1|C(group)[2]`.

This would be particularly useful for teaching and learning purposes.

**Comment by tomicapretto** [comment]
Interesting idea indeed!

**Comment by tomicapretto** [comment]
@lucianopaz that is correct. Now that you did the work, I'm going to add it as a method :)
